### PR TITLE
encoder: Add rot13 example script & SHA256 Predefined

### PR DESCRIPTION
**Pull request by kingthorin**
- Added a ROT13 example script.
- Added a SHA256 predefined processor.

Part of: zaproxy/zaproxy#5996

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

File: addOns/encoder/src/main/zapHomeFiles/scripts/templates/encode-decode/rot13.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+/**
+ * Process the input value and return the encoded/decoded/hashed etc. value
+ *
+ * Use EncodeDecodeResult.withError("Error Description") to provide an error description
+ * inside the result view
+ *
+ * @param {String} value - The input value
+ * @returns {EncodeDecodeResult} - The value that was encoded/decoded/hashed etc.
+ */
```
Comment (thc202) [inline addOns/encoder/src/main/zapHomeFiles/scripts/templates/encode-decode/rot13.js:12]: Could be removed (same as template).

**Comment by thc202** [comment]
Could you update the changelog?

**Comment by kingthorin** [comment]
Done & done.

**Comment by thc202** [comment]
Thank you!

**Pull request merged**